client.go

```diff
@@ -79,9 +79,9 @@ func (c *Client) fullURL(suffix string) string {
 
 type requestSetter func(req *http.Request)
 
-func withBetaVersion(version string) requestSetter {
+func withBetaVersion(version BetaVersion) requestSetter {
```

Comment:
I have a suggestion. According to the Claude documentation: `anthropic-beta: If passing in multiple, use a comma separated list without spaces, e.g. beta1,beta2.` Here, the parameters can be restructured as: `func withBetaVersion(versions ...BetaVersion) requestSetter.`